runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowPipelineDebugOptions.java

```diff
@@ -178,6 +180,20 @@ public Dataflow create(PipelineOptions options) {
   void setDumpHeapOnOOM(boolean dumpHeapBeforeExit);
 
   /**
+   * CAUTION: This option implies dumpHeapOnOOM, and has similar caveats. Specifically, heap
+   * dumps can of comparable size to the default boot disk. Consider increasing the boot disk size
+   * before setting this flag to true.
+   */
+  @Description(
+      "[EXPERIMENTAL] Set to a GCS bucket (directory) to upload heap dumps to the given location.\n"
+      + "Enabling this implies that heap dumps should be generated on OOM (--dumpHeapOnOOM=true)\n"
+      + "Uploads will continue until the pipeline is stopped or updated without this option.\n")
+  @Experimental
+  String getSaveHeapDumpsToGcsPath();
+  @Experimental
```

Comment:
Is it customary to add the `@Experimental` attribute to getter and setter? I see others apply to the block as a whole: https://github.com/apache/beam/blob/master/runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowPipelineOptions.java#L96

Reply:
I think most places only apply it to the getter, but that seems incorrect wrong. The `@Experimental` annotation applies to a specific method, and indicates that that method is subject to change. If we apply it only to a getter, does that mean that we cannot change the setter?

Reply:
I think it is good to have it on both, in case we actually manage to automate tooling around API compatibility. It does look redundant, but I think that is just a fact of life.

Reply:
I don't have a strong opinion here, but it's worth noting that while `PipelineOptions` are specified using java constructs, the actual implementation uses crazy reflection magic in [`PipelineOptionsFactory`](https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java) and [`ProxyInvocationHandler`](https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/options/ProxyInvocationHandler.java) to emulate real Java objects.

We validate that `@JsonIgnore` and `@Default` are [only annotated on getters and not setters](https://github.com/apache/beam/blob/b75fe358e257b8f2bf096698a4336c1d32aa44eb/sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java#L1040). And `@Default` annotations are [only read from the getter](https://github.com/apache/beam/blob/b75fe358e257b8f2bf096698a4336c1d32aa44eb/sdks/java/core/src/main/java/org/apache/beam/sdk/options/ProxyInvocationHandler.java#L156). Same with [`@Description` and `@Validation.Requred`](https://github.com/apache/beam/blob/b75fe358e257b8f2bf096698a4336c1d32aa44eb/sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java#L725).

So on one hand, to be consistent with `PipelineOptions` annotations it should only be specified on the getter. But `@Experimental` is presumably for IDEs/static analysis/API compatibility testing, which would favor specifying on each method.

Reply:
Removed the line in favor of consistency.